broker-daemon/utils/exponential-backoff.js

```diff
@@ -0,0 +1,19 @@
+const logger = require('./logger')
+const delay = require('./delay')
+
+async function exponentialBackoff (callFunction, attempts, delayTime) {
+  try {
+    var res = callFunction()
+  } catch (error) {
+    logger.error(`Error (${error}) with ${callFunction}, retry attempts left: ${attempts}`)
+    if (attempts > 0) {
+      await delay(delayTime)
+      exponentialBackoff(callFunction, --attempts, delayTime * 2)
```

Comment:
need to assign the return value here for `res`, otherwise you won't get the return value